### PR TITLE
User add new spot

### PR DESCRIPTION
**Pull request by danielmdavis**
Add and edit spots from appropriate pages and routes. with tests. associated minor tweaks, esp. in tables

### CODE REVIEW

File: app/controllers/spots_controller.rb [hunk 1]
```diff
@@ -1,6 +1,58 @@
...
+
   def index
   end
 
+  def show
+  end
+
+  def new
+    @spot = Spot.new
+  end
```
Comment (nwalberts) [inline app/controllers/spots_controller.rb]: Add a space here

File: app/controllers/spots_controller.rb [hunk 2]
```diff
@@ -1,6 +1,58 @@
...
+
+  protected
+  def spot_params
+    params.require(:spot).permit(:name, :location, :photo, :artist, :description)
+  end
+
+  def authorize_user
+    if !user_signed_in?
+      # raise ActionController::RoutingError.new("Not Found ipsum")
```
Comment (nwalberts) [inline app/controllers/spots_controller.rb]: remove comment

File: app/views/spots/edit.html.erb [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+      <%= f.text_field :description %>
+    </div>
+
+    <div class="actions">
+      <%= f.submit "Update", :class => "button form-submit" %>
+    </div>
+  <% end %>
+
+  </div>
+</div>
```
Comment (nwalberts) [inline app/views/spots/edit.html.erb]: Please do not use `=>` (arrow) syntax, is this is very old school and deprecated. Usually you can use normal key/value pair syntax

File: app/views/spots/new.html.erb [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        <%= f.text_field :artist %>
+      </div>
+
+      <div class="actions">
+        <%= f.submit "Add Spot", :class => "button form-submit" %>
+      </div>
+    <% end %>
+
+  </div>
+<div>
```
Comment (nwalberts) [inline app/views/spots/new.html.erb]: Update to remove arrow syntax

File: db/seeds.rb [hunk 5]
```diff
@@ -1,20 +1,21 @@
...
 #
-# Examples:
-#
-#   movies = Movie.create([{ name: 'Star Wars' }, { name: 'Lord of the Rings' }])
-#   Character.create(name: 'Luke', movie: movies.first)
+
+user_1 = User.create!(:role => 'member', :email => 'nick@gmail.com', :password => 'topsecret', :password_confirmation => 'topsecret')
+user_2 = User.create!(:role => 'member', :email => 'brianna@gmail.com', :password => 'topsecret', :password_confirmation => 'topsecret')
+user_3 = User.create!(:role => 'member', :email => 'amylynn@gmail.com', :password => 'topsecret', :password_confirmation => 'topsecret')
+admin = User.create!(:role => 'admin', :email => 'ateam@gmail.com', :password => 'topsecret', :password_confirmation => 'topsecret')
```
Comment (nwalberts) [inline db/seeds.rb]: Please update to remove arrow syntax in favor of key/value pair syntax

File: spec/features/user_adds_new_spot_spec.rb [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+feature 'authenticated user adds new spot' do
+  before(:each) do
+    user = FactoryBot.create(:user)
+    visit new_user_session_path
+
+    fill_in 'Email', with: user.email
+    fill_in 'Password', with: user.password
+
+    click_button 'Log in'
+  end
```
Comment (nwalberts) [inline spec/features/user_adds_new_spot_spec.rb]: You can skip this by using the `login_in` method. 
https://github.com/plataformatec/devise/wiki/How-To:-Test-with-Capybara
Or configure your own `sing_in` helper method for tests

Comment (nwalberts) [reply spec/features/user_adds_new_spot_spec.rb]: But I will leave that up to you

File: spec/features/user_adds_new_spot_spec.rb [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+    # fill_in 'Location', with: "Boston, MA"
+
+    # click_button 'Add Spot'
+
+    # visit spot_path
+
+    # expect(page).to have_content('Cool new art spot')
+    # expect(page).to have_content('Lorem ipsum dolum sita met.')
+    # expect(page).to have_content('Boston, MA')
+  end
```
Comment (nwalberts) [inline spec/features/user_adds_new_spot_spec.rb]: remove commented out code.....or better yet finish this test

File: db/seeds.rb [hunk 8]
```diff
@@ -1,15 +1,39 @@
...
+  { name: 'Lorem ipsum',location:'1 Cambridge Sq, Cambridge MA',description: 'abe is the man', photo: 'https://cdn-www.enfocus.com/sites/combell-www.enfocus.com/files/media/blog/2017-08-09-Lorem-Ipsum/lorem-ipsum.jpg', user: user_3 },
+  { name: 'Christian the Pirate',location:'Deck 7 Common, Boston MA',description: 'fastest eyes', user: user_3 },
+  { name: 'Evan the Great',location:'1 Cambridge Sq, Cambridge MA',description: 'lorem ipsum dolum', user: user_3 },
+  { name: 'Content from admin',location:'1 Cambridge Sq, Cambridge MA',description: 'lorem ipsum dolum', user: admin },
 ]
 
 spots.each do |item|
   spot = Spot.create!(item)
 end
+<<<<<<< HEAD
```
Comment (AL6981) [inline db/seeds.rb]: This needs to come out.

**Pull request merged**